### PR TITLE
feat: Support reading private key from string

### PR DESCRIPTION
**Pull request by patriknordlen**
#### Summary

This PR addresses #13555 by adding a `private_key` configuration key that can be used to read the private key from a string instead of a file.

### CODE REVIEW

File: plugins/source/github/client/client.go [hunk 1]
```diff
@@ -81,7 +82,12 @@ func New(ctx context.Context, logger zerolog.Logger, spec Spec) (schema.ClientMe
 
 	ghServices := map[string]GithubServices{}
 	for _, auth := range spec.AppAuth {
-		k, err := key.FromFile(auth.PrivateKeyPath)
+		var k *rsa.PrivateKey
```
[the remark this answers is not in the document]
Comment (patriknordlen) [reply plugins/source/github/client/client.go]: Thanks for pushing a fix! The PR is ready for review.

File: plugins/source/github/client/spec.go [hunk 2]
```diff
@@ -47,8 +48,11 @@ func (s *Spec) Validate() error {
 		if appAuth.Org == "" {
 			return fmt.Errorf("missing org in app auth configuration")
 		}
-		if appAuth.AppID != "" && appAuth.PrivateKeyPath == "" {
-			return fmt.Errorf("missing private key path in configuration")
+		if appAuth.AppID != "" && (appAuth.PrivateKeyPath == "" && appAuth.PrivateKey == "") {
+			return fmt.Errorf("missing private key specification in configuration")
```
Comment (erezrokah) [inline plugins/source/github/client/spec.go:52]:
```suggestion
			return fmt.Errorf("missing private key specification in configuration. Please specify using either `private_key` or `private_key_path`")
```

File: plugins/source/github/client/spec.go [hunk 3]
```diff
@@ -47,8 +48,11 @@ func (s *Spec) Validate() error {
...
+		}
+		if appAuth.AppID != "" && (appAuth.PrivateKeyPath != "" && appAuth.PrivateKey != "") {
+			return fmt.Errorf("both private key and private key path specified in configuration")
```
Comment (erezrokah) [inline plugins/source/github/client/spec.go:55]:
```suggestion
			return fmt.Errorf("both private key and private key path specified in configuration. Please remove the configuration for either `private_key_path` or `private_key`")
```

**Review by erezrokah — CHANGES_REQUESTED** [review]
Thanks for following up @patriknordlen, added a couple of comments regarding the error messages

**Comment by patriknordlen** [comment]
> Thanks for following up @patriknordlen, added a couple of comments regarding the error messages

Thanks for the suggestions, accepted and committed!

**Review by erezrokah — APPROVED** [review]

**Pull request merged**